test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala

```diff
@@ -0,0 +1,359 @@
+          _.set(IsOtherOfficialNinoPage(2), false)).flatMap(
+          _.set(OtherOfficialsPassportPage(2), Passport("GB12345", "GB", LocalDate.of(year, month, dayOfMonth)))).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(2), true)).flatMap(
+          _.set(OtherOfficialPreviousAddressLookupPage(2), ConfirmedAddressConstants.address)).flatMap(
+          _.set(AddAnotherOtherOfficialPage, true)).success.value) mustBe true
+      }
+
+      "return true with three other officials with passport, Nino and previous address (Scenario 5)" in {
+        helper.checkComplete(commonAllThree.set(IsOtherOfficialNinoPage(0), false).flatMap(
+          _.set(OtherOfficialsPassportPage(0), Passport("GB12345", "GB", LocalDate.of(year, month, dayOfMonth)))).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(0), true)).flatMap(
+          _.set(OtherOfficialPreviousAddressLookupPage(0), ConfirmedAddressConstants.address)).flatMap(
+          _.set(IsOtherOfficialNinoPage(1), true)).flatMap(
+          _.set(OtherOfficialsNinoPage(1), "AA123456A")).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(1), true)).flatMap(
+          _.set(OtherOfficialPreviousAddressLookupPage(1), ConfirmedAddressConstants.address)).flatMap(
+          _.set(IsOtherOfficialNinoPage(2), true)).flatMap(
+          _.set(OtherOfficialsNinoPage(2), "AA123456A")).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(2), true)).flatMap(
+          _.set(OtherOfficialPreviousAddressLookupPage(2), ConfirmedAddressConstants.address)).flatMap(
+          _.set(AddAnotherOtherOfficialPage, true)).success.value) mustBe true
+      }
+
+      "return false with incorrect data for two other officials" in {
+        helper.checkComplete(commonFirstTwo.set(IsOtherOfficialNinoPage(0), true).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(0), false)).flatMap(
+          _.set(IsOtherOfficialNinoPage(1), false)).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(1), false)).success.value) mustBe false
+      }
+
+      "return false with correct data for two other officials, but not correct data for the 3rd other official" in {
+        helper.checkComplete(commonFirstTwo.set(IsOtherOfficialNinoPage(0), true).flatMap(
+          _.set(OtherOfficialsNinoPage(0), "AA123456A")).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(0), false)).flatMap(
+          _.set(IsOtherOfficialNinoPage(1), true)).flatMap(
+          _.set(OtherOfficialsNinoPage(1), "AA123456A")).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(1), false)).flatMap(
+          _.set(AddAnotherOtherOfficialPage, true)).success.value) mustBe false
+      }
+
+      "return false with incorrect data for three other officials" in {
+        helper.checkComplete(commonAllThree.set(IsOtherOfficialNinoPage(0), false).flatMap(
+          _.set(OtherOfficialsPassportPage(0), Passport("GB12345", "GB", LocalDate.of(year, month, dayOfMonth)))).flatMap(
```

Comment:
Here in this class lot of step are repeating for for creating user data. 
Could it be possible to define complete data set with 2 other officials and 3 other officials, then removing the data from as per the test requirement.

Reply:
on it.